compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -2082,12 +2082,16 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                 })
                 .collect();
 
-            let end = if candidates.len() <= 9 { candidates.len() } else { 8 };
+            let end = if candidates.len() <= 9 || self.tcx.sess.opts.verbose {
```

Comment:
I can't help but point out that this original logic was/is a little goofy, in that it says: "if the length is 9 or less, keep it. If its greater than 9, then cut it to length 8."

Why wouldn't you cut it to length 9? (Or use "8 or less"  as the relevant threshold.)

(My current guess is that it was an off-by-one error in some past person's end in terms of whether `candidates[..end]` would include or exclude the `end` itself.)

Reply:
I think they might've wanted to avoid a message like `and 1 others` which would've needed some rewording such as `and 1 other`(lacking an `s` ending), so then avoid the more complicated logic with that extra `s`.

Thus, show 8 + the 2 others msg if above 9, or show all 9 or less with no msg. Makes some sense :D and is kinda clever.


```rust
error[E0277]: the trait bound `String: Reconcile` is not satisfied
  --> src/main.rs:72:13
   |
72 |     process(value); // This line will cause a compilation error
   |     ------- ^^^^^ the trait `Reconcile` is not implemented for `String`
   |     |
   |     required by a bound introduced by this call
   |
   = help: the following other types implement trait `Reconcile`:
             bool
             i128
             i16
             i32
             i64
             i8
             u128
             u16
           and 2 others
note: required by a bound in `process`
```

```
rror[E0277]: the trait bound `String: Reconcile` is not satisfied
  --> /home/user/sandbox/rust/05_sandbox/error/and_17_others/src/main.rs:72:13
   |
72 |     process(value); // This line will cause a compilation error
   |     ------- ^^^^^ the trait `Reconcile` is not implemented for `String`
   |     |
   |     required by a bound introduced by this call
   |
   = help: the following other types implement trait `Reconcile`:
             bool
             i8
             i16
             i32
             i64
             i128
             u8
             u16
             u32
             u128
note: required by a bound in `process`
```

Reply:
It wasn't to avoid the "1 others" message, but rather if we're already "wasting" a line to say there's another one, then just print it.